src/functionalTest/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/PostTaskSearchControllerTest.java

```diff
@@ -235,31 +262,38 @@ public void should_return_a_200_with_empty_search_results_user_jurisdiction_perm
             new SearchParameter(STATE, SearchOperator.IN, singletonList("unassigned"))
         ));
 
+        common.setupOrganisationalRoleAssignmentWithCustomAttributes(
+            authenticationHeaders,
+            Map.of(
+                "primaryLocation", "17595",
+                "jurisdiction", "IA"
+            )
+        );
+
         Response result = restApiActions.post(
             ENDPOINT_BEING_TESTED,
             searchTaskRequest,
-            authorizationHeadersProvider.getTribunalCaseworkerBAuthorization()
+            authenticationHeaders
         );
 
         result.then().assertThat()
             .statusCode(HttpStatus.OK.value())
             .body("tasks.size()", equalTo(0));
 
-        tasksCreated.stream()
-            .map(map -> map.get("taskId")).forEach(taskId -> common.cleanUpTask(taskId, REASON_COMPLETED));
+        tasksCreated.forEach(task -> common.cleanUpTask(task.getTaskId(), REASON_COMPLETED));
 
     }
 
```

Comment:
Do we also need a task_by_id_with_restricted_role_assignment() test for PostTaskSearchController endpoint?